editoast/src/views/pathfinding/electrical_profiles.rs

```diff
@@ -0,0 +1,324 @@
+use crate::error::{InternalError, Result};
+use crate::models::{
+    pathfinding::Pathfinding, ElectricalProfileSet, LightRollingStockModel, Retrieve,
+};
+use crate::schema::electrical_profiles::ElectricalProfileSetData;
+use crate::views::electrical_profiles::ElectricalProfilesError;
+use crate::views::pathfinding::{
+    rangemap_utils::{make_path_range_map, Float, RangedValue},
+    PathfindingError,
+};
+use crate::DbPool;
+use actix_web::{
+    dev::HttpServiceFactory,
+    get, services,
+    web::{Data, Json, Path, Query},
+};
+use rangemap::RangeMap;
+use serde::{Deserialize, Serialize};
+use std::collections::{HashMap, HashSet};
+use std::fmt::Debug;
+
+pub fn routes() -> impl HttpServiceFactory {
+    services![electrical_profiles_on_path,]
+}
+
+/// Build a rangemap for each track section, giving the electrical profile for each range
+fn map_electrical_profiles(
+    electrical_profile_set: ElectricalProfileSetData,
+    power_class: String,
+    track_ids: HashSet<String>,
+) -> (HashMap<String, RangeMap<Float, String>>, Vec<InternalError>) {
+    let mut warnings = Vec::new();
+
+    let mut res = HashMap::new();
+    for profile in electrical_profile_set.levels {
+        if profile.power_class != power_class {
+            continue;
+        }
+        let mut overlapping_ranges = Vec::new();
+        for track_range in &profile.track_ranges {
+            if !track_ids.contains(track_range.track.as_str()) {
+                continue;
+            }
+            let res_entry = res
+                .entry(track_range.track.to_string())
+                .or_insert_with(RangeMap::new);
+            let range = track_range.begin.into()..track_range.end.into();
+            if res_entry.overlaps(&range) {
+                overlapping_ranges.push(track_range.clone());
+            }
+            res_entry.insert(range, profile.value.clone());
+        }
+        if !overlapping_ranges.is_empty() {
+            warnings
+                .push(PathfindingError::ElectricalProfilesOverlap { overlapping_ranges }.into());
+        }
+    }
+    (res, warnings)
+}
+
+#[derive(Debug, Deserialize)]
+struct ProfilesOnPathQuery {
+    rolling_stock_id: i64,
+    electrical_profile_set_id: i64,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ProfilesOnPathResponse {
+    electrical_profile_ranges: Vec<RangedValue>,
+    warnings: Vec<InternalError>,
+}
+
+#[get("/{path_id}/electrical_profiles")]
+async fn electrical_profiles_on_path(
```

Comment:
Could you add an `utoipa::path` annotation on this endpoint. Even if you don't use routes! and the annotation is left unused, that'll make the migration to utoipa easier